tests/meltano/core/test_plugin_invoker.py

```diff
@@ -167,3 +176,19 @@ async def test_expand_nonpip_command_exec_args(
 
         assert "VIRTUAL_ENV" not in env
         assert "PYTHONPATH" not in env
+
+    def test_cwd(self, alternate_cwd_plugin_invoker):
+        root = alternate_cwd_plugin_invoker.project.root
+        assert alternate_cwd_plugin_invoker.cwd() == root / "path"
+        assert alternate_cwd_plugin_invoker.cwd(command="relative") == root / "path2"
+        assert alternate_cwd_plugin_invoker.cwd(command="absolute") == Path("/root")
```

Comment:
Heads up - we run tests on windows now as well, which are currently failing:

https://github.com/meltano/meltano/runs/7313447720?check_suite_focus=true#step:9:648

```
_________________________ TestPluginInvoker.test_cwd __________________________
self = <test_plugin_invoker.TestPluginInvoker object at 0x000001DAA9868310>
alternate_cwd_plugin_invoker = <meltano.core.plugin_invoker.PluginInvoker object at 0x000001DAADEFCBB0>
    def test_cwd(self, alternate_cwd_plugin_invoker):
        root = alternate_cwd_plugin_invoker.project.root
        assert alternate_cwd_plugin_invoker.cwd() == root / "path"
        assert alternate_cwd_plugin_invoker.cwd(command="relative") == root / "path2"
>       assert alternate_cwd_plugin_invoker.cwd(command="absolute") == Path("/root")
E       AssertionError: assert WindowsPath('C:/root') == WindowsPath('/root')
E        +  where WindowsPath('C:/root') = <bound method PluginInvoker.cwd of <meltano.core.plugin_invoker.PluginInvoker object at 0x000001DAADEFCBB0>>(command='absolute')
E        +    where <bound method PluginInvoker.cwd of <meltano.core.plugin_invoker.PluginInvoker object at 0x000001DAADEFCBB0>> = <meltano.core.plugin_invoker.PluginInvoker object at 0x000001DAADEFCBB0>.cwd
E        +  and   WindowsPath('/root') = Path('/root')
D:\a\meltano\meltano\tests\meltano\core\test_plugin_invoker.py:184: AssertionError
```